llvm/lib/LTO/LTOBackend.cpp

```diff
@@ -221,8 +221,18 @@ createTargetMachine(const Config &Conf, const Target *TheTarget, Module &M) {
   else
     CodeModel = M.getCodeModel();
 
+  TargetOptions TargetOpts = Conf.Options;
+
+  if (TargetOpts.MCOptions.ABIName.empty()) {
+    Metadata *ABI = M.getModuleFlag("target-abi");
```

Comment:
I'm not sure what the current status is of PR100833, but note it adds a facility to get this module flag to Module.h. Consider adding that here so they are commoned.

Reply:
#100833 seems need few more days, will rebase and use that once that patch landed.